libutils/string_lib.c

```diff
@@ -593,6 +594,107 @@ long StringToLongExitOnError(const char *str)
     return result;
 }
 
+/**
+ * @brief Converts a string of numerals in base 10 to 64-bit signed int
+ *
+ * Result is stored in *value_out, return value should be checked.
+ * On error *value_out is unmodified and false is returned.
+ *
+ * @see StringToLong()
+ * @param[in] str String with numerals to convert, cannot be NULL
+ * @param[out] value_out Where to store result on success, cannot be NULL
+ * @return 0 on success, error code otherwise (see source code)
+ */
+int StringToInt64(const char *str, int64_t *value_out)
+{
+    nt_static_assert(sizeof(int64_t) == sizeof(intmax_t));
+    nt_static_assert(ERANGE != 0);
+    assert(str != NULL);
+    assert(value_out != NULL);
+
+    char *endptr = NULL;
+    int64_t val;
+
+    errno = 0;
+    val = strtoimax(str, &endptr, 10);
+
+    if ((errno == ERANGE && (val == LONG_MAX || val == LONG_MIN)))
```

Comment:
Should be `INTMAX_MAX` and `INTMAX_MIN`.

Reply:
Thanks, fixed.